server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/IMAPServerFactory.java

```diff
@@ -36,37 +39,45 @@
 import org.apache.james.protocols.lib.netty.AbstractServerFactory;
 
 import com.github.fge.lambdas.functions.ThrowingFunction;
+import com.google.common.collect.ImmutableSet;
 
 public class IMAPServerFactory extends AbstractServerFactory {
 
     protected final FileSystem fileSystem;
     protected final ThrowingFunction<HierarchicalConfiguration<ImmutableNode>, ImapSuite> imapSuiteProvider;
     protected final ImapMetrics imapMetrics;
     protected final GaugeRegistry gaugeRegistry;
+    protected final ConnectionCheckFactory connectionCheckFactory;
 
     @Inject
     @Deprecated
     public IMAPServerFactory(FileSystem fileSystem, ImapDecoder decoder, ImapEncoder encoder, ImapProcessor processor,
-                             MetricFactory metricFactory, GaugeRegistry gaugeRegistry) {
+                             MetricFactory metricFactory, GaugeRegistry gaugeRegistry, ConnectionCheckFactory connectionCheckFactory) {
         this.fileSystem = fileSystem;
+        this.connectionCheckFactory = connectionCheckFactory;
         this.imapSuiteProvider = any -> new ImapSuite(decoder, encoder, processor);
         this.imapMetrics = new ImapMetrics(metricFactory);
         this.gaugeRegistry = gaugeRegistry;
     }
 
     public IMAPServerFactory(FileSystem fileSystem, ThrowingFunction<HierarchicalConfiguration<ImmutableNode>, ImapSuite> imapSuiteProvider,
-                             MetricFactory metricFactory, GaugeRegistry gaugeRegistry) {
+                             MetricFactory metricFactory, GaugeRegistry gaugeRegistry, ConnectionCheckFactory connectionCheckFactory) {
         this.fileSystem = fileSystem;
         this.imapSuiteProvider = imapSuiteProvider;
         this.imapMetrics = new ImapMetrics(metricFactory);
         this.gaugeRegistry = gaugeRegistry;
+        this.connectionCheckFactory = connectionCheckFactory;
     }
 
     protected IMAPServer createServer(HierarchicalConfiguration<ImmutableNode> config) {
         ImapSuite imapSuite = imapSuiteProvider.apply(config);
-        return new IMAPServer(imapSuite.getDecoder(), imapSuite.getEncoder(), imapSuite.getProcessor(), imapMetrics, gaugeRegistry);
+        ImmutableSet<String> connectionChecks = Arrays.stream(config.getStringArray("additionalConnectionChecks")).collect(ImmutableSet.toImmutableSet());
+
+        return new IMAPServer(imapSuite.getDecoder(), imapSuite.getEncoder(), imapSuite.getProcessor(), imapMetrics, gaugeRegistry, connectionCheckFactory.create(ImapConfiguration.builder()

```

Comment:
IMO
```
ImapConfiguration.builder()
            .connectionChecks(connectionChecks)
            .build()
```
it looks quite mock,  
- It should be full (real) "ImapConfiguration", not just a empty ImapConfiguration with only `connectionChecks` value
- or the input of `ConnectionCheckFactory::create` should be `HierarchicalConfiguration<ImmutableNode> config` or another

Reply:
I prefer the option 2